extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/MqttChannelTypeProvider.java

```diff
@@ -28,14 +28,17 @@
 import org.eclipse.smarthome.core.thing.type.ChannelType;
 import org.eclipse.smarthome.core.thing.type.ChannelTypeProvider;
 import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.osgi.service.component.annotations.Component;
 
 /**
  * If the user configures a channel and defines for example minimum/maximum values, we need a specific
+ * channel type provider. This one is started on-demand only, as soon as {@link MqttBrokerHandlerFactory} requires it.
  *
  * @author David Graeff - Initial contribution
  *
  */
 @NonNullByDefault
+@Component(immediate = false, service = ChannelTypeProvider.class)
```

Comment:
this got to be `service = { ChannelTypeProvider.class, MqttChannelTypeProvider.class}` because `MqttBrokerHandlerFactory` has a reference to `MqttChannelTypeProvider`.